### ISSUE TITLE
valid? resets input parameter

### ISSUE THREAD

**Issue by martinmalek** [carr/phone#53]
Whenever I send a parameter which is not a valid phone number
Phoner::Phone.valid?(myParam) it is reset and returned as ""

Example:
myParam = "ABC123"
Phoner::Phone.valid?(myParam) => false
myParam is now ""

**Comment by elskwid** [comment]
@hoguej Took me long enough!!!

I don't believe that the parameter modification is an issue. There are tests to explicitly test for the modification when using `#valid?` and I just added one (a6e6ab0ce00) to make sure `#parse` is safe.

**Comment by calebanderson** [comment]
For anyone who is using an older version of the gem (or until this is fixed in the most recent release) this worked for me.
`````
module Phoner
  class Phone
    def self.parse_with_duplication(string, options = {})
      parse_without_duplication(string.dup, options)
    end

    singleton_class.alias_method_chain :parse, :duplication
  end
end
`````

**Comment by elskwid** [comment]
@calebanderson have you tried using the beta?

**Comment by calebanderson** [comment]
I haven't, but the `sub!` in line makes me think it could still be an issue:
https://github.com/carr/phone/blob/v1.3.0.beta0/lib/phone.rb#L192

**Comment by elskwid** [comment]
@calebanderson good call! I fixed that and never pushed out a new release. I can release a new beta later - any chance you can try with master?

**Comment by calebanderson** [comment]
@elskwid Sure I'll give it a shot. I've already looked over the code quite a bit and everything looked good.

**Comment by calebanderson** [comment]
@elskwid Everything I checked out seems to be working great, at least as far as I can tell given my little knowledge about international phone numbers.

**Comment by elskwid** [comment]
@calebanderson I will push out a new version as soon as I get a chance (tonight or tomorrow morning). Will keep you posted. Thanks so much!!

**Comment by elskwid** [comment]
@calebanderson took longer than expected (what doesn't??!?!) but v1.3.0.beta1 is out for you. If it works I'll push out the final version in the days to come.
